src/settings-ui/Microsoft.PowerToys.Settings.UI/Controls/KeyVisual/KeyVisual.cs

```diff
@@ -2,29 +2,169 @@
+            IsEnabledChanged -= KeyVisual_IsEnabledChanged;
+            _keyVisual = (KeyVisual)this;
+            _keyPresenter = (ContentPresenter)_keyVisual.GetTemplateChild(KeyPresenter);
+            Update();
+            SetEnabledState();
+            SetErrorState();
+            IsEnabledChanged += KeyVisual_IsEnabledChanged;
             base.OnApplyTemplate();
         }
+
+        private static void OnContentChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            ((KeyVisual)d).Update();
+        }
+
+        private static void OnSizeChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            ((KeyVisual)d).Update();
+        }
+
+        private static void OnIsErrorChanged(DependencyObject d, DependencyPropertyChangedEventArgs e)
+        {
+            ((KeyVisual)d).SetErrorState();
+        }
+
+        private void Update()
+        {
+            if (_keyVisual == null)
+            {
+                return;
+            }
+
+            if (_keyVisual.Content != null)
+            {
+                if (_keyVisual.Content.GetType() == typeof(string))
+                {
+                    _keyVisual.Style = GetStyleSize("TextKeyVisualStyle");
+                    _keyVisual._keyPresenter.Content = _keyVisual.Content;
+                }
+                else
+                {
+                    _keyVisual.Style = GetStyleSize("IconKeyVisualStyle");
+
+                    switch ((int)_keyVisual.Content)
+                    {
+                        /* We can enable other glyphs in the future
+                        case 13: // The Enter key or button.
+                            _keyVisual._keyPresenter.Content = "\uE751"; break;
+
+                        case 8: // The Back key or button.
+                            _keyVisual._keyPresenter.Content = "\uE750"; break;
+
+                        case 16: // The right Shift key or button.
+                        case 160: // The left Shift key or button.
+                        case 161: // The Shift key or button.
+                            _keyVisual._keyPresenter.Content = "\uE752"; break; */
+
+                        case 38: _keyVisual._keyPresenter.Content = "\uE0E4"; break; // The Up Arrow key or button.
+                        case 40: _keyVisual._keyPresenter.Content = "\uE0E5"; break; // The Down Arrow key or button.
+                        case 37: _keyVisual._keyPresenter.Content = "\uE0E2"; break; // The Left Arrow key or button.
+                        case 39: _keyVisual._keyPresenter.Content = "\uE0E3"; break; // The Right Arrow key or button.
+
+                        case 91: // The left Windows key
+                        case 92: // The right Windows key
+                            PathIcon winIcon = XamlReader.Load(@"<PathIcon xmlns=""http://schemas.microsoft.com/winfx/2006/xaml/presentation"" Data=""M9,17V9h8v8ZM0,17V9H8v8ZM9,8V0h8V8ZM0,8V0H8V8Z"" />") as PathIcon;
```

Comment:
Is it possible to use the icon U+E782 from HoloLens MDL2 font?

Reply:
Yeah, was thinking the same thing. However, I'm not sure if HoloLens MDL2 is a default font installed on all Windows machines. I'm also not sure if we can just embed it into PowerToys, or if there would be specific licensing constraints. Therefore, this is the safer option for now. I hope #216 will be picked up at some point - then we can just stick to Segoe Fluent Icons.

Reply:
#216 seems to be a number typo mistake. Is it?

Reply:
> not sure if HoloLens MDL2 is a default font installed on all Windows machines

Yes, it is.
![image](https://user-images.githubusercontent.com/65828559/131831056-fbbb76a3-f9b8-4936-a004-c75bce8726f6.png)

https://docs.microsoft.com/en-us/typography/font-list/hololens-mdl2-assets

Reply:
> #216 seems to be a number typo mistake. Is it?

Yes and no. It's a different repo: https://github.com/microsoft/fluentui-system-icons/issues/216